### PR TITLE
YETUS-896 Add an emoji column to the github vote table

### PR DESCRIPTION
**Pull request by infraio**

### CODE REVIEW

**Comment by apache-yetus** [comment]
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://builds.apache.org/job/yetus-github-multibranch/job/PR-67/1/console in case of problems.

**Comment by apache-yetus** [comment]
:+1: **+1 overall**






| Emoji | Vote | Subsystem | Runtime | Comment |
|:----:|:----:|----------:|--------:|:--------|
| :+1: | 0 | reexec | 1205 | Docker mode activated. |
|||| _ Prechecks _ |
| :+1: | +1 | dupname | 0 | No case conflicting files found. |
| :+1: | +1 | @author | 0 | The patch does not contain any @author tags. |
|||| _ master Compile Tests _ |
|||| _ Patch Compile Tests _ |
| :+1: | +1 | shellcheck | 0 | There were no new shellcheck issues. |
| :+1: | +1 | shelldocs | 0 | There were no new shelldocs issues. |
| :+1: | +1 | whitespace | 0 | The patch has no whitespace issues. |
|||| _ Other Tests _ |
| :+1: | +1 | asflicense | 10 | The patch does not generate ASF License warnings. |
| :+1: | | | 1231 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.0 Server=19.03.0 base: https://builds.apache.org/job/yetus-github-multibranch/job/PR-67/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/yetus/pull/67 |
| Optional Tests | dupname asflicense shellcheck shelldocs |
| uname | Linux 91aca168d630 4.4.0-139-generic #165-Ubuntu SMP Wed Oct 24 10:58:50 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | precommit/src/main/shell/personality/yetus.sh |
| git revision | master / 6b94d9d |
| Max. process+thread count | 46 (vs. ulimit of 2000) |
| modules | C: precommit U: precommit |
| Console output | https://builds.apache.org/job/yetus-github-multibranch/job/PR-67/1/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.6.0 |
| Powered by | Apache Yetus 0.11.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

**Comment by busbey** [comment]
The vote column seems redundant with the emoji maybe?

Is it worth a cli option to pick between emoji + text vote or just text vote or just emoji for votes?

Also should 0 votes have no emoji or perhaps a different one?

**Comment by infraio** [comment]
> The vote column seems redundant with the emoji maybe?
> 
> Is it worth a cli option to pick between emoji + text vote or just text vote or just emoji for votes?
> 
> Also should 0 votes have no emoji or perhaps a different one?

I prefer to only use emoji. It is enough to show the vote result.

**Comment by busbey** [comment]
Okay so how about an option that determines if the vote column is emoji or text?

File: precommit/src/main/shell/test-patch.d/github.sh [hunk 1]
```diff
@@ -610,9 +610,11 @@ function github_finalreport
     comment=$(echo "${ourstring}"  | cut -f5 -d\|)
 
     if [[ "${vote}" = "H" ]]; then
-      echo "||| _${comment}_ |" >> "${commentfile}"
+      echo "|||| _${comment}_ |" >> "${commentfile}"
+    elif [[ "${vote}" = "-1" ]]; then
+      echo "| :broken_heart: ${TP_VOTE_TABLE[${i}]}" >> "${commentfile}"
     else
-      echo "${TP_VOTE_TABLE[${i}]}" >> "${commentfile}"
+      echo "| :+1: ${TP_VOTE_TABLE[${i}]}" >> "${commentfile}"
```
Comment (aw-was-here) [inline precommit/src/main/shell/test-patch.d/github.sh]: There are three other values that need handling here, +0, -0, and 0.  Marking them with the same emoji as +1 is very misleading.

Comment (infraio) [reply precommit/src/main/shell/test-patch.d/github.sh]: If we use different emoji for these vote. I thought we should use emoji + text vote both.

**Comment by infraio** [comment]
The complete list of github markdown emoji markup can be found in https://gist.github.com/rxaviers/7360908

**Comment by infraio** [comment]
> Okay so how about an option that determines if the vote column is emoji or text?

Add a with-emoji option, if with-emoji, then use emoji and vote text both. If not, just use the vote text.

**Comment by apache-yetus** [comment]
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://builds.apache.org/job/yetus-github-multibranch/job/PR-67/2/console in case of problems.

**Comment by apache-yetus** [comment]
:+1: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 2016 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
||| _ master Compile Tests _ |
||| _ Patch Compile Tests _ |
| +1 | shellcheck | 1 | There were no new shellcheck issues. |
| +1 | shelldocs | 1 | There were no new shelldocs issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
||| _ Other Tests _ |
| +1 | asflicense | 10 | The patch does not generate ASF License warnings. |
| | | 2045 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.0 Server=19.03.0 base: https://builds.apache.org/job/yetus-github-multibranch/job/PR-67/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/yetus/pull/67 |
| Optional Tests | dupname asflicense shellcheck shelldocs |
| uname | Linux 3a3de9025db3 4.15.0-54-generic #58-Ubuntu SMP Mon Jun 24 10:55:24 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | precommit/src/main/shell/personality/yetus.sh |
| git revision | master / 6b94d9d |
| Max. process+thread count | 41 (vs. ulimit of 2000) |
| modules | C: precommit U: precommit |
| Console output | https://builds.apache.org/job/yetus-github-multibranch/job/PR-67/2/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.6.0 |
| Powered by | Apache Yetus 0.11.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

**Comment by infraio** [comment]
@busbey  @aw-was-here  Any more concerns?

File: precommit/src/main/shell/test-patch.d/github.sh [hunk 2]
```diff
@@ -88,6 +90,10 @@ function github_parse_args
         delete_parameter "${i}"
         GITHUB_USER=${i#*=}
       ;;
+      --with-emoji)
```
Comment (busbey) [inline precommit/src/main/shell/test-patch.d/github.sh]: Should match the pattern of starting with GitHub, so `--github-with-emoji`

**Comment by apache-yetus** [comment]
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://builds.apache.org/job/yetus-github-multibranch/job/PR-67/3/console in case of problems.

**Comment by apache-yetus** [comment]
:+1: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 1870 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
||| _ master Compile Tests _ |
||| _ Patch Compile Tests _ |
| +1 | shellcheck | 0 | There were no new shellcheck issues. |
| +1 | shelldocs | 1 | There were no new shelldocs issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
||| _ Other Tests _ |
| +1 | asflicense | 9 | The patch does not generate ASF License warnings. |
| | | 1897 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.0 Server=19.03.0 base: https://builds.apache.org/job/yetus-github-multibranch/job/PR-67/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/yetus/pull/67 |
| Optional Tests | dupname asflicense shellcheck shelldocs |
| uname | Linux 950bccfc6c13 4.15.0-52-generic #56-Ubuntu SMP Tue Jun 4 22:49:08 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | precommit/src/main/shell/personality/yetus.sh |
| git revision | master / 6b94d9d |
| Max. process+thread count | 46 (vs. ulimit of 2000) |
| modules | C: precommit U: precommit |
| Console output | https://builds.apache.org/job/yetus-github-multibranch/job/PR-67/3/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.6.0 |
| Powered by | Apache Yetus 0.11.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

**Comment by infraio** [comment]
Ping @busbey

**Comment by Apache9** [comment]
I agree that the emoji is duplicated with the vote. I suggested that, we can introduce an option called `--github-use-emoji`, then we will just use the emoji instead of the old +1, -1 0, -0 and so on.

This is my implelmentation.

https://github.com/apache/hbase/commit/8e99b85718a2cd76103c535e5ba71afabf75b482

I tried to use it in hbase first but it does not work...

**Comment by busbey** [comment]
Closed in favor of #68

**Pull request closed without merging**